fvm/crypto/crypto.go

```diff
@@ -52,19 +66,35 @@ func (DefaultSignatureVerifier) Verify(
 	return valid, nil
 }
 
-// NewHasher returns a crypto hasher supported by runtime.
-func NewHasher(hashAlgo hash.HashingAlgorithm) hash.Hasher {
+func Hash(hashAlgo hash.HashingAlgorithm, tag string, data []byte) ([]byte, error) {
```

Comment:
It would be clearer to separate the standard hash from the "tagged" hash. 
```suggestion
func HashWithTag(hashAlgo hash.HashingAlgorithm, tag string, data []byte) ([]byte, error) {
```